pkg/endpoint/endpoint.go

```diff
@@ -1662,6 +1662,15 @@ func (e *Endpoint) RunMetadataResolver(bwm bandwidth.Manager, resolveMetadata Me
 			Group: resolveLabelsControllerGroup,
 			DoFunc: func(ctx context.Context) error {
 				ns, podName := e.GetK8sNamespace(), e.GetK8sPodName()
+				if ns == "" || podName == "" {
```

Comment:
Why not use `e.K8sNamespaceAndPodNameIsSet()`? We already have this check at the call site, why is this needed here?

https://github.com/cilium/cilium/blob/6feffd1ea1c091fc33b96ac80e50bb0ec570d1c0/daemon/cmd/endpoint.go#L467